### PR TITLE
Bratseth/errors

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Review by hmusum — DISMISSED** [review]

File: config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java [hunk 1]
```diff
@@ -34,84 +34,74 @@ private RankProfile makeRankProfile(Schema schema, String name) {
...
-        }
-        for (var value : parsed.getRankFeatures()) {
-            profile.addRankFeatures(value);
+        try {
+            RankProfile profile = makeRankProfile(schema, parsed.name());
+            parsed.getInherited().forEach(profile::inherit);
+            parsed.isStrict().ifPresent(profile::setStrict);
+            parsed.isUseSignificanceModel().ifPresent(profile::setUseSignificanceModel);
+            parsed.getConstants().values().forEach(profile::add);
+            ;
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java:43]: Stray ;

File: config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java [hunk 2]
```diff
@@ -34,84 +34,74 @@ private RankProfile makeRankProfile(Schema schema, String name) {
...
-        parsed.getGlobalPhaseExpression().ifPresent(profile::setGlobalPhaseRanking);
-        parsed.getGlobalPhaseRerankCount().ifPresent(profile::setGlobalPhaseRerankCount);
-
-        for (var value : parsed.getMatchFeatures()) {
-            profile.addMatchFeatures(value);
-        }
-        for (var value : parsed.getRankFeatures()) {
-            profile.addRankFeatures(value);
+        try {
+            RankProfile profile = makeRankProfile(schema, parsed.name());
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java:38]: Perhaps separate conversion and error handling with a private conversion function.
Less indentation, and error handling is easier to follow.

```
private RankProfile createRankProfile(Schema schema, ParsedRankProfile parsed) {
    RankProfile profile = makeRankProfile(schema, parsed.name());
    ......
    return profile;
}

void convertRankProfile(Schema schema, ParsedRankProfile parsed) {
    try {
        rankProfileRegistry.add(createRankProfile(schema, parsed));
    } catch (IllegalArgumentException e) {
         throw new IllegalArgumentException("In " + parsed, e);
    }
}
```

**Review by baldersheim — DISMISSED** [review]
Semantics looks fine.

File: config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java (hunk 2 above)
Comment (bratseth) [reply config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java]: Okay

**Pull request merged**